CMakeLists.txt

```diff
@@ -450,12 +450,13 @@ resolve_dependency(glog)
 set_source(fmt)
 resolve_dependency(fmt 9.0.0)
 
+find_package(lz4 REQUIRED)
```

Comment:
will this make lz4 a required dependency for the entire library? 

If we are now building the API, could we make sure we build it in a pluggable way (similar to how Arrow does it)?

Reply:
Yes. I agree, if it's possible keeping this optional by building the compression api only on demand would be preferable.